pkg/kubelet/kubelet.go

```diff
@@ -1894,7 +1904,12 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	}
 
 	// Mount volumes and update the volume manager
-	podVolumes, err := kl.mountExternalVolumes(pod)
+	// Default limits for containers here to have downward API expose user-friendly limits to pods.
+	defaultedPod, err := kl.defaultPodLimitsForDownwardApi(pod)
+	if err != nil {
+		return err
+	}
+	podVolumes, err := kl.mountExternalVolumes(defaultedPod)
```

Comment:
ok i think here kl.mountExternalVolumes(defaultedPod) will take care of passing updated copy of the pod to downward api volumes. So should be fine. Although it seems to send updated copy to all volumes, may be its ok.